csrc/options.h

```diff
@@ -0,0 +1,207 @@
+                       //! path and replay result
+  Cubin, //! Dump compiled CUBIN
+  Sass, // Dump disassembled SASS
+  Ptx, //! Dump compiled PTX
+  BankConflictInfo, //! Dump bank confliction info
+  SyncMap, //! RAW dependency info
+  LowerVerbose, //! Print all passes' transform in GpuLower::lower
+  ExprSimplification, //! Print all passes' transform in simplifyExpr
+  ExprSort, //! Print merging decisions on expression sorting
+  LoopRotation, //! Print loop rotation log
+  MatmulChecks, //! Print logs from tools around matmul scheduler used in
+                //! segmenter
+  Occupancy, // Dump occupancy
+  IndexType, //! Print the index type of the launched kernel
+  EndOfOption //! Placeholder for counting the number of elements
+};
+
+//! Types of features to enable
+//!
+//! These can be set through the `PYTORCH_NVFUSER_ENABLE` environment variable
+//!
+enum class EnableOption {
+  Complex, //! Enable complex support on python
+  ConvDecomposition, //! Enable conv-bias decomposition
+  GraphOp, //! Enable graphOps(index_select/gather/scatter)
+  KernelDb, //! Enable Kernel Database
+  KernelProfile, //! Enable intra-kernel performance profiling
+  LinearDecomposition, //! Enable linear-bias decomposition
+  WarnRegisterSpill, //! Enable warnings of register spill
+  EndOfOption //! Placeholder for counting the number of elements
+};
+
+//! Types of features to disable
+//!
+//! These can be set through the `PYTORCH_NVFUSER_DISABLE` environment variable
+//!
+enum class DisableOption {
+  CompileToSass, //! Disable direct compilation to sass so the ptx can be
+                 //! examined
+  ExprSimplify, //! Disable expression simplifier
+  Fallback, //! Disable fallback
+  Fma, //! Disable FMA instructions
+  GroupedGridWelfordOuterOpt, //! Disable use of outer-optimized
+                              //! grouped grid welford kernel
+  IndexHoist, //! Disable index hoisting
+  MagicZero, //! Disable nvfuser_zero
+  Nvtx, //! Disable NVTX instrumentation
+  PredicateElimination, //! Disable predicate elimination
+  VarNameRemapping, //! Disable variable name remapping
+  WelfordVectorization, //! Disable vectorizaton of Welford ops
+  EndOfOption //! Placeholder for counting the number of elements
+};
+
+//! The base template class for the options such as EnableOption
+template <typename OptionEnum>
+class Options {
+ public:
+  Options() = default;
+
+  bool has(OptionEnum option) const {
+    return options_.count(option);
+  }
+
+  const std::vector<std::string>& getArgs(OptionEnum option) const {
+    TORCH_INTERNAL_ASSERT(has(option), "Option not set");
+    return options_.at(option);
+  }
+
+  void set(OptionEnum option_type, std::vector<std::string> option = {}) {
+    options_[option_type] = option;
+  }
+
+  void unset(OptionEnum option_type) {
+    options_.erase(option_type);
+  }
+
+  static std::unordered_map<OptionEnum, std::vector<std::string>>
+  getOptionsFromEnv();
```

Comment:
We are not calling `getOptionsFromEnv` at all. So we are just basically not reading anything from env variable.

Reply:
linking #510

Reply:
😮  Sorry about that. Will fix it ASAP